sustainability/migrations/17.0.1.1.0/pre-migration.py

```diff
@@ -0,0 +1,46 @@
+from openupgradelib import openupgrade
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    openupgrade.rename_models(
+        env.cr,
+        [("carbon.factor.source", "carbon.factor.database")],
+    )
+    openupgrade.rename_tables(
+        env.cr,
+        [("carbon_factor_source", "carbon_factor_database")],
+    )
+    openupgrade.logged_query(
+        env.cr,
+        """
+        UPDATE ir_model_fields
+        SET name = 'carbon_database_id'
+        WHERE name = 'carbon_source_id'
+        AND model = 'carbon.factor';
+        """,
+    )
```

Comment:
Isn't this already done in the rename_fields ?
https://github.com/OCA/openupgradelib/blob/c5713fcbaabe00d4ca0d1c9426103dd862943237/openupgradelib/openupgrade.py#L684